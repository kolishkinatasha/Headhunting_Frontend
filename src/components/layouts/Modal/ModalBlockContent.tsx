import React from 'react';
import { Row, Typography } from '@ff/ui-kit';
import { observer } from 'mobx-react';

import classes from './Modal.module.scss';

type ModalBlockContentProps = {
  title: string;
};

const ModalBlockContent: React.FC<ModalBlockContentProps> = observer(
  ({ title, children }) => (
    <div className={classes.block}>
      <Typography.Title level={3} className={classes.title}>
        {title}
      </Typography.Title>
      <Row className={classes.block_content}>{children}</Row>
    </div>
  ),
);
export default ModalBlockContent;
