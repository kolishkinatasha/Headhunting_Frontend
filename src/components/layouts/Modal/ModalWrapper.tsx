import { Icon, Modal } from '@ff/ui-kit';
import { observer } from 'mobx-react';
import React from 'react';

import classes from './Modal.module.scss';

type ModalWrapperProps = {
  button: any;
};

const ModalWrapper: React.FC<ModalWrapperProps> = observer(
  ({ button, children }) => {
    const [isOpen, setIsOpen] = React.useState(false);

    const openModal = () => {
      setIsOpen(true);
    };

    const closeModal = () => {
      setIsOpen(false);
    };
    return (
      <div className={classes.modalBox}>
        <div
          role="button"
          onClick={openModal}
          onKeyDown={openModal}
        >{button}
        </div>
        <Modal
          className={classes.modal}
          onBackdropClick={closeModal}
          visible={isOpen}
          width="1200px"
        >
          <div onClick={closeModal} onKeyDown={openModal} role="button" className={classes.close_button}>
            <Icon name="close" />
          </div>
          {children}
        </Modal>
      </div>
    );
  },
);

export default ModalWrapper;
