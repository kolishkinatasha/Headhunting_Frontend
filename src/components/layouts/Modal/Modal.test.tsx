import React from 'react';
import { render } from '@testing-library/react';

import ModalWrapper from './ModalWrapper';
import ModalBlockContent from './ModalBlockContent';

describe('ModalWrapper', () => {
  it('should be the same', () => {
    const app = render(<ModalWrapper button={<button>Button</button>} />);
    expect(app).toMatchSnapshot();
  });
});
describe('ModalBlockContent', () => {
  it('should be the same', () => {
    const app = render(<ModalBlockContent title="Title" />);
    expect(app).toMatchSnapshot();
  });
});
