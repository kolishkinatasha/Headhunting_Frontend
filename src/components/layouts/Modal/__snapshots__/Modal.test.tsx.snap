// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBlockContent should be the same 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      style="position: absolute; top: 0px; left: 0px; width: 100%;"
    />
    <div>
      <div
        class="block"
      >
        <h3
          class="fns-typography fns-typography_h3 title"
        >
          Title
        </h3>
        <div
          class="fns-row block_content"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="block"
    >
      <h3
        class="fns-typography fns-typography_h3 title"
      >
        Title
      </h3>
      <div
        class="fns-row block_content"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ModalWrapper should be the same 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modalBox"
      >
        <div
          role="button"
        >
          <button>
            Button
          </button>
        </div>
      </div>
    </div>
    <div
      style="position: absolute; top: 0px; left: 0px; width: 100%;"
    />
  </body>,
  "container": <div>
    <div
      class="modalBox"
    >
      <div
        role="button"
      >
        <button>
          Button
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
